Description: Fix for imlib2 update as imlib_strerror() has been introduced to imlib2, so changing source files afected.
Origin: http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/x11/eterm/patches/patch-src_pixmap.c?r1=1.1
Last-Update: 2023-03-10 
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/pixmap.c
+++ b/src/pixmap.c
@@ -172,7 +172,7 @@
 
 #ifdef PIXMAP_SUPPORT
 const char *
-imlib_strerror(Imlib_Load_Error err)
+eterm_imlib_strerror(Imlib_Load_Error err)
 {
     switch (err) {
         case IMLIB_LOAD_ERROR_NONE:
@@ -1447,7 +1447,7 @@
         if (f) {
             im = imlib_load_image_with_error_return(f, &im_err);
             if (!im) {
-                libast_print_error("Unable to load image file \"%s\" -- %s\n", file, imlib_strerror(im_err));
+                libast_print_error("Unable to load image file \"%s\" -- %s\n", file, eterm_imlib_strerror(im_err));
                 return 0;
             } else {
                 reset_simage(simg, (RESET_IMLIB_IM | RESET_PMAP_PIXMAP | RESET_PMAP_MASK));
@@ -2178,7 +2178,7 @@
 
             temp_im = imlib_load_image_with_error_return(icon_path, &im_err);
             if (!temp_im) {
-                libast_print_error("Unable to load icon file \"%s\" -- %s\n", icon_path, imlib_strerror(im_err));
+                libast_print_error("Unable to load icon file \"%s\" -- %s\n", icon_path, eterm_imlib_strerror(im_err));
             } else {
                 /* If we're going to render the image anyway, might as well be nice and give it to the WM in a size it likes. */
                 if (XGetIconSizes(Xdisplay, Xroot, &icon_sizes, &count)) {
--- a/src/pixmap.h
+++ b/src/pixmap.h
@@ -221,7 +221,7 @@
 extern const char *get_image_type(unsigned char);
 extern unsigned char image_mode_any(unsigned char);
 #ifdef PIXMAP_SUPPORT
-extern const char *imlib_strerror(Imlib_Load_Error);
+extern const char *eterm_imlib_strerror(Imlib_Load_Error);
 extern unsigned short parse_pixmap_ops(char *);
 extern unsigned short set_pixmap_scale(const char *, pixmap_t *);
 extern unsigned char check_image_ipc(unsigned char);
